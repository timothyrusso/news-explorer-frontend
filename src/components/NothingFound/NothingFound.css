.nothing-found {
  background-color: #F5F6F7;
  width: 100%;
  min-height: 374px;
}

.nothing-found__icon {
  display: block;
  margin: 0 auto;
  padding: 80px 0 24px;
  max-width: 74px;
}

.nothing-found__title {
  margin: 0;
  text-align: center;
  padding-bottom: 16px;
  font-family: 'Roboto Slab';
  font-style: normal;
  font-weight: 400;
  font-size: 26px;
  line-height: 1.15;
  color: #1A1B22;
}

.nothing-found__text {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 356px;
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 1.33;
  color: #B6BCBF;
  text-align: center;
  padding: 0 40px;
}
