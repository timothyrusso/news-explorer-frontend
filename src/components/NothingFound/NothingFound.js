import "./NothingFound.css";
import nothingFoundIcom from "../../images/not-found.svg";

const NothingFound = () => {
  return (
    <div className="nothing-found">
      <img
        src={nothingFoundIcom}
        alt="Nothing found icon"
        className="nothing-found__icon"
      ></img>
      <h2 className="nothing-found__title">Nothing found</h2>
      <p className="nothing-found__text">
        Sorry, but nothing matched your search terms.
      </p>
    </div>
  );
};

export default NothingFound;
