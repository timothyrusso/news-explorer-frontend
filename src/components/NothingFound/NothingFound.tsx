import React, { FC } from 'react';
import './NothingFound.css';

type NothingFoundProps = {
  title: string;
  text: string;
  nothingFoundIcon: string;
};

const NothingFound: FC<NothingFoundProps> = ({
  title,
  text,
  nothingFoundIcon,
}) => {
  return (
    <div className="nothing-found">
      <img
        src={nothingFoundIcon}
        alt="Nothing found icon"
        className="nothing-found__icon"
      ></img>
      <h2 className="nothing-found__title">{title}</h2>
      <p className="nothing-found__text">{text}</p>
    </div>
  );
};

export default NothingFound;
