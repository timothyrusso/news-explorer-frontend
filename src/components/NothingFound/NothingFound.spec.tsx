import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import NothingFound from './NothingFound';

describe('NothingFound component', () => {
  it('Should render the NothingFound component', () => {
    const { asFragment } = render(
      <Router>
        <NothingFound
          title="Nothing found"
          text="Sorry, but nothing matched your search terms."
          nothingFoundIcon="nothingFoundIcon_path"
        />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
