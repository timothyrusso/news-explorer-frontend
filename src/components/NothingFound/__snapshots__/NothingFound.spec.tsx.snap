// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NothingFound component Should render the NothingFound component 1`] = `
<DocumentFragment>
  <div
    class="nothing-found"
  >
    <img
      alt="No results for Nothing found"
      class="nothing-found__icon"
      src="nothingFoundIcon_path"
    />
    <h2
      class="nothing-found__title"
    >
      Nothing found
    </h2>
    <p
      class="nothing-found__text"
    >
      Sorry, but nothing matched your search terms.
    </p>
  </div>
</DocumentFragment>
`;
